libres/lib/job_queue/lsf_driver.cpp

```diff
@@ -959,24 +927,61 @@ void lsf_driver_free_job(void *__job) {
     lsf_job_free(job);
 }
 
+namespace lsf {
+std::vector<std::string> split_string(std::string str, std::string delimiter) {
+    std::vector<std::string> splitted = {};
+    size_t pos = 0, next_pos = 0;
+
+    while ((next_pos = str.find(delimiter, pos)) != std::string::npos) {
+        splitted.push_back(str.substr(pos, next_pos - pos));
+        pos = next_pos + delimiter.size();
+    }
+    splitted.push_back(str.substr(pos, str.size() - pos));
+    return splitted;
+}
+
+/*
+ * Parses the given file containing colon separated hostnames, ie.
+ * "hname1:hname2:hname3". This is the same format as
+ * written by lsf_job_write_bjobs_to_file().
+ */
+std::vector<std::string> parse_hostnames(const char *fname) {
+    std::ifstream stream(fname);
+
+    std::string line;
+    if (std::getline(stream, line)) { //just read the first line
+        // bjobs uses : as delimiter
+        std::vector<std::string> hosts_w_star = split_string(line, ":");
+
+        // change "4*be-lsf01" to "be-lsf01"
```

Comment:
I don't understand the significance of `[number]*` in lsf hostnames. What does that mean? (Also maybe add as comment)

Reply:
The `hostnames` variable originates in LSF, where the value corresponds to PE's in a parallel process setting. I.e. if you have a parallel program with ten processes running on host `compute0` and 6 processes running on host `compute2` you would start with the string: `10*compute0:6*compute2`